Update crates to build on nightly

itowlson opened the pull request:
Fixes #90.

lann reviewed (approved):
I didn't even consider "just update everything" :facepalm:

itowlson commented:
It looks like another PR (possibly #93) also updated Cargo.lock in a way that when I rebased this there were no outstanding changes.  @lann `main` now builds on nightly for me - please yell out if it doesn't for you!
